# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/HttpChannelChild.cpp
# Commit: d59b42f58f86
# Full Hash: d59b42f58f861e049c1270035ea7568acc0605b9
# Author: Junior Hsu <juhsu@mozilla.com>
# Date: 2020-06-30 10:00:42
# Regressor Bug: 1633935
# File Overlap Count: 1
# Description:
#   Bug 1633935 - P3 send OS*R and ODA via pHttpChannel for multipart channel, r=mayhemer,necko-reviewers
#   
#   Depends on D76969
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D76970
# ==============================================================================

diff -r cbcd3ee87d28 -r d59b42f58f86 netwerk/protocol/http/HttpChannelChild.cpp
--- a/netwerk/protocol/http/HttpChannelChild.cpp	Tue Jun 30 00:32:38 2020 +0000
+++ b/netwerk/protocol/http/HttpChannelChild.cpp	Tue Jun 30 00:32:53 2020 +0000
@@ -401,9 +401,6 @@
     const HttpChannelOnStartRequestArgs& aArgs) {
   LOG(("HttpChannelChild::ProcessOnStartRequest [this=%p]\n", this));
   MOZ_ASSERT(OnSocketThread());
-  MOZ_ASSERT(!mMultiPartID,
-             "Should only send OnStartRequest on the main-thread channel when "
-             "using multi-part!");
   MOZ_RELEASE_ASSERT(!mFlushedForDiversion,
                      "Should not be receiving any more callbacks from parent!");
 
@@ -448,14 +445,17 @@
     // We don't need to notify the background channel if this is a multipart
     // stream, since all messages will be sent over the main-thread IPDL in
     // that case.
-    if (mBgChild && !aArgs.multiPartID()) {
-      MOZ_RELEASE_ASSERT(gSocketTransportService);
-      DebugOnly<nsresult> rv = gSocketTransportService->Dispatch(
-          NewRunnableMethod(
-              "HttpBackgroundChannelChild::OnStartRequestReceived", mBgChild,
-              &HttpBackgroundChannelChild::OnStartRequestReceived),
-          NS_DISPATCH_NORMAL);
-    }
+
+    // TODO: Never reach here. Remove later.
+
+    // if (mBgChild && !aArgs.multiPartID()) {
+    //   MOZ_RELEASE_ASSERT(gSocketTransportService);
+    //   DebugOnly<nsresult> rv = gSocketTransportService->Dispatch(
+    //       NewRunnableMethod(
+    //           "HttpBackgroundChannelChild::OnStartRequestReceived", mBgChild,
+    //           &HttpBackgroundChannelChild::OnStartRequestReceived),
+    //       NS_DISPATCH_NORMAL);
+    // }
   }
 
   return IPC_OK();
@@ -778,9 +778,6 @@
     const uint64_t& aOffset, const uint32_t& aCount, const nsCString& aData) {
   LOG(("HttpChannelChild::ProcessOnTransportAndData [this=%p]\n", this));
   MOZ_ASSERT(OnSocketThread());
-  MOZ_ASSERT(
-      !mMultiPartID,
-      "Should only send ODA on the main-thread channel when using multi-part!");
   MOZ_RELEASE_ASSERT(!mFlushedForDiversion,
                      "Should not be receiving any more callbacks from parent!");
   mEventQ->RunOrEnqueue(
@@ -1009,9 +1006,6 @@
     const nsTArray<ConsoleReportCollected>& aConsoleReports) {
   LOG(("HttpChannelChild::ProcessOnStopRequest [this=%p]\n", this));
   MOZ_ASSERT(OnSocketThread());
-  MOZ_ASSERT(
-      !mMultiPartID,
-      "Should only send ODA on the main-thread channel when using multi-part!");
   MOZ_RELEASE_ASSERT(!mFlushedForDiversion,
                      "Should not be receiving any more callbacks from parent!");
 